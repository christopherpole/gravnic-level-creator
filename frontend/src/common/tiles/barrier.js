import React from 'react';
import PropTypes from 'prop-types';

const BarrierTile = ({ color, powered }) => (
  <svg width="100%" height="100%" viewBox="0 0 100 100" version="1.1">
    <defs />
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g id="aaa">
        <rect fill="#fff" fillRule="evenodd" x="0" y="0" width="100" height="100" />
        {powered && <rect fill={color} fillRule="evenodd" x="25" y="25" width="50" height="50" />}
        <rect stroke={color} strokeWidth="2" x="13.5" y="13.5" width="74" height="74" />
      </g>
    </g>
  </svg>
);

BarrierTile.defaultProps = {
  color: 'black',
};

BarrierTile.propTypes = {
  color: PropTypes.string,
  powered: PropTypes.bool.isRequired,
};

export default BarrierTile;
