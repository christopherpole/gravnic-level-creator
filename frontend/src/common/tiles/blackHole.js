import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const Wrapper = styled.div`
  background: white;
  height: 100%;
  width: 100%;

  &:before {
    transition: all 0.2s linear;
    height: 50%;
    width: 50%;
    background: black;
    border-radius: 100%;
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    margin-top: -25%;
    margin-left: -25%;
  }

  ${props =>
    props.shrinking &&
    css`
      &:before {
        height: 0;
        width: 0;
      }
    `};
`;

const BlackHoleTile = ({ shrinking }) => <Wrapper shrinking={shrinking} />;

BlackHoleTile.defaultProps = {
  shrinking: false,
};

BlackHoleTile.propTypes = {
  shrinking: PropTypes.bool,
};

export default BlackHoleTile;
