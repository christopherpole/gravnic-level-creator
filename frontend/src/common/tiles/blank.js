import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  background: ${props => props.theme.backgroundColor};
`;

const BlankTile = () => <Wrapper />;

export default BlankTile;
