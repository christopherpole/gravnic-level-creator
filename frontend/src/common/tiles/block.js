import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  background: ${props => (props.shrinking ? 'white' : props.color)};
  transition: background ${props => props.moveSpeed}ms linear;
  height: 100%;
  width: 100%;
`;

const BlockTile = ({ color, shrinking, moveSpeed }) => (
  <Wrapper color={color} moveSpeed={moveSpeed} shrinking={shrinking} />
);

BlockTile.defaultProps = {
  shrinking: false,
  moveSpeed: 0,
};

BlockTile.propTypes = {
  shrinking: PropTypes.bool,
  moveSpeed: PropTypes.number,
  color: PropTypes.string.isRequired,
};

export default BlockTile;
