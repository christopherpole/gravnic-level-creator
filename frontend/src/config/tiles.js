import { ENTITIES, MOVE_NONE, MOVE_UP, MOVE_RIGHT, MOVE_DOWN, MOVE_LEFT } from 'gravnic-game';

export default [
  {
    id: '1',
    entity: {
      entityId: ENTITIES.NONE.id,
    },
    name: 'Empty',
  },
  {
    id: '2',
    entity: {
      entityId: ENTITIES.FLOOR.id,
    },
    name: 'Floor',
  },
  {
    id: '3',
    entity: {
      entityId: ENTITIES.GLASS.id,
    },
    name: 'Glass',
  },
  {
    id: '4',
    entity: {
      entityId: ENTITIES.BLOCK.id,
      color: '#ff0000',
    },
    name: 'Red block',
  },
  {
    id: '5',
    entity: {
      entityId: ENTITIES.BLOCK.id,
      color: '#FF8C00',
    },
    name: 'Orange block',
  },
  {
    id: '6',
    entity: {
      entityId: ENTITIES.BLOCK.id,
      color: '#FFFF00',
    },
    name: 'Yellow block',
  },
  {
    id: '7',
    entity: {
      entityId: ENTITIES.BLOCK.id,
      color: '#008000',
    },
    name: 'Green block',
  },
  {
    id: '8',
    entity: {
      entityId: ENTITIES.BLOCK.id,
      color: '#0000FF',
    },
    name: 'Blue block',
  },
  {
    id: '9',
    entity: {
      entityId: ENTITIES.BLOCK.id,
      color: '#800080',
    },
    name: 'Purple block',
  },
  {
    id: '10',
    entity: {
      entityId: ENTITIES.BLOCK.id,
      color: '#8B4513',
    },
    name: 'Brown block',
  },
  {
    id: '11',
    entity: {
      entityId: ENTITIES.BLOCK.id,
      color: '#FFE4E1',
    },
    name: 'Pink block',
  },
  {
    id: '12',
    entity: {
      entityId: ENTITIES.RAINBOW_BLOCK.id,
    },
    name: 'Rainbow block',
  },
  {
    id: '13',
    entity: {
      entityId: ENTITIES.BLACK_HOLE.id,
    },
    name: 'Black hole',
  },
  {
    id: '14',
    entity: {
      entityId: ENTITIES.STICKY_SPOT.id,
    },
    name: 'Sticky spot',
  },
  {
    id: '15',
    entity: {
      entityId: ENTITIES.LAVA.id,
    },
    name: 'Lava',
  },
  {
    id: '16',
    entity: {
      entityId: ENTITIES.SMART_BOMB.id,
    },
    name: 'Smart bomb',
  },
  {
    id: '17',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#ff0000',
      },
    },
    name: 'Color changer (red block)',
  },
  {
    id: '18',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#FF8C00',
      },
    },
    name: 'Color changer (orange block)',
  },
  {
    id: '19',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#FFFF00',
      },
    },
    name: 'Color changer (yellow block)',
  },
  {
    id: '20',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#008000',
      },
    },
    name: 'Color changer (green block)',
  },
  {
    id: '21',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#0000FF',
      },
    },
    name: 'Color changer (blue block)',
  },
  {
    id: '22',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#800080',
      },
    },
    name: 'Color changer (purple block)',
  },
  {
    id: '23',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#8B4513',
      },
    },
    name: 'Color changer (brown block)',
  },
  {
    id: '24',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#FFE4E1',
      },
    },
    name: 'Color changer (pink block)',
  },
  {
    id: '25',
    entity: {
      entityId: ENTITIES.COLOR_CHANGER.id,
      targetEntity: {
        entityId: ENTITIES.RAINBOW_BLOCK.id,
      },
    },
    name: 'Color changer (rainbow block)',
  },
  {
    id: '26',
    entity: {
      entityId: ENTITIES.GRAVITY_CHANGER.id,
      direction: MOVE_NONE,
    },
    name: 'Gravity changer (static)',
  },
  {
    id: '27',
    entity: {
      entityId: ENTITIES.GRAVITY_CHANGER.id,
      direction: MOVE_UP,
    },
    name: 'Gravity changer (up)',
  },
  {
    id: '28',
    entity: {
      entityId: ENTITIES.GRAVITY_CHANGER.id,
      direction: MOVE_RIGHT,
    },
    name: 'Gravity changer (right)',
  },
  {
    id: '29',
    entity: {
      entityId: ENTITIES.GRAVITY_CHANGER.id,
      direction: MOVE_DOWN,
    },
    name: 'Gravity changer (down)',
  },
  {
    id: '30',
    entity: {
      entityId: ENTITIES.GRAVITY_CHANGER.id,
      direction: MOVE_LEFT,
    },
    name: 'Gravity changer (left)',
  },
  {
    id: '31',
    entity: {
      entityId: ENTITIES.CRATE.id,
    },
    name: 'Crate',
  },
  {
    id: '32',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      powered: true,
    },
    name: 'Barrier',
  },
  {
    id: '33',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#ff0000',
      powered: true,
    },
    name: 'Barrier (red)',
  },
  {
    id: '34',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#FF8C00',
      powered: true,
    },
    name: 'Barrier (orange)',
  },
  {
    id: '35',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#FFFF00',
      powered: true,
    },
    name: 'Barrier (yellow)',
  },
  {
    id: '36',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#008000',
      powered: true,
    },
    name: 'Barrier (green)',
  },
  {
    id: '37',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#0000FF',
      powered: true,
    },
    name: 'Barrier (blue)',
  },
  {
    id: '38',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#800080',
      powered: true,
    },
    name: 'Barrier (purple)',
  },
  {
    id: '39',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#8B4513',
      powered: true,
    },
    name: 'Barrier (brown)',
  },
  {
    id: '40',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#FFE4E1',
      powered: true,
    },
    name: 'Barrier (pink)',
  },
  {
    id: '41',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      powered: false,
    },
    name: 'Barrier (unpowered)',
  },
  {
    id: '42',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#ff0000',
      powered: false,
    },
    name: 'Barrier (red, unpowered)',
  },
  {
    id: '43',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#FF8C00',
      powered: false,
    },
    name: 'Barrier (orange, unpowered)',
  },
  {
    id: '44',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#FFFF00',
      powered: false,
    },
    name: 'Barrier (yellow, unpowered)',
  },
  {
    id: '45',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#008000',
      powered: false,
    },
    name: 'Barrier (green, unpowered)',
  },
  {
    id: '46',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#0000FF',
      powered: false,
    },
    name: 'Barrier (blue, unpowered)',
  },
  {
    id: '47',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#800080',
      powered: false,
    },
    name: 'Barrier (purple, unpowered)',
  },
  {
    id: '48',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#8B4513',
      powered: false,
    },
    name: 'Barrier (brown, unpowered)',
  },
  {
    id: '49',
    entity: {
      entityId: ENTITIES.BARRIER.id,
      color: '#FFE4E1',
      powered: false,
    },
    name: 'Barrier (pink, unpowered)',
  },
  {
    id: '50',
    entity: {
      entityId: ENTITIES.CRUSHER.id,
    },
    name: 'Crusher',
  },
];
