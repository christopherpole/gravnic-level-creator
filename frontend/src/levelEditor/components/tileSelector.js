import React from 'react';
import styled, { css } from 'styled-components';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';

import Tile from 'common/tile';
import { selectTile } from '../actions';

export const Wrapper = styled.div`
  position: relative;
  flex-grow: 1;
`;

export const WrapperInner = styled.div`
  border: 1px solid ${props => props.theme.foregroundColor};
  padding: ${props => props.theme.structureSpacing};
  display: flex;
  flex-wrap: wrap;
  overflow-y: scroll;
  align-content: flex-start;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`;

export const TileWrapper = styled.div`
  width: 25%;
  height: 0;
  padding-bottom: 25%;
  position: relative;
  cursor: pointer;

  ${props =>
    props.isSelected &&
    css`
      &:before {
        border: 2px solid yellow;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        position: absolute;
        content: '';
        z-index: 2;
      }
    `};
`;

export const TileSelector = ({ selectedTileId, selectTileAction, availableTiles }) => (
  <Wrapper id="tile-selector">
    <WrapperInner>
      {availableTiles.map(tile => (
        <TileWrapper
          className="tile"
          isSelected={tile.id === selectedTileId}
          onClick={() => {
            selectTileAction(tile.id);
          }}
          key={tile.id}
        >
          <Tile tileId={tile.id} />
        </TileWrapper>
      ))}
    </WrapperInner>
  </Wrapper>
);

TileSelector.defaultProps = {
  selectedTileId: null,
};

TileSelector.propTypes = {
  selectedTileId: PropTypes.string,
  selectTileAction: PropTypes.func.isRequired,
  availableTiles: PropTypes.array.isRequired,
};

const mapStateToProps = state => ({
  selectedTileId: state.levelEditor.selectedTileId,
  availableTiles: state.levelEditor.availableTiles,
});

const mapDispatchToProps = dispatch => ({
  selectTileAction: bindActionCreators(selectTile, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TileSelector);
