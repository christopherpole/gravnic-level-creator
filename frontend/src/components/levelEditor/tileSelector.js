import React from 'react';
import styled, { css } from 'styled-components';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';

import { selectTile } from '../../actions/levelEditor';
import Tile from '../common/tile';
import tiles from '../../config/tiles.json';

export const Wrapper = styled.div`
  position: relative;
  grid-row: 1 / 3;
  grid-column: 2 / 3;
`;

export const WrapperInner = styled.div`
  border: 1px solid ${props => props.theme.foregroundColor};
  padding: ${props => props.theme.structureSpacing};
  display: flex;
  flex-wrap: wrap;
  overflow-y: scroll;
  align-content: flex-start;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`;

export const TileWrapper = styled.div`
  width: 25%;
  height: 0;
  padding-bottom: 25%;
  position: relative;
  cursor: pointer;

  ${props =>
    props.isSelected &&
    css`
      &:before {
        border: 2px solid yellow;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        position: absolute;
        content: '';
        z-index: 2;
      }
    `};
`;

export const TileSelector = ({ selectedTileId, selectTileAction }) => (
  <Wrapper id="tile-selector">
    <WrapperInner>
      {tiles.map(tile => (
        <TileWrapper
          className="tile"
          isSelected={tile.id === selectedTileId}
          onClick={() => {
            selectTileAction(tile.id);
          }}
          key={tile.id}
        >
          <Tile tileId={tile.id} isSelected={tile.id === selectedTileId} />
        </TileWrapper>
      ))}
    </WrapperInner>
  </Wrapper>
);

TileSelector.defaultProps = {
  selectedTileId: null,
};

TileSelector.propTypes = {
  selectedTileId: PropTypes.number,
  selectTileAction: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  selectedTileId: state.levelEditor.selectedTileId,
});

const mapDispatchToProps = dispatch => ({
  selectTileAction: bindActionCreators(selectTile, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(TileSelector);
