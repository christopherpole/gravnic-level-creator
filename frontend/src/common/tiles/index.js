import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ENTITIES } from 'gravnic-game';
import { connect } from 'react-redux';

import { getEntityForTileId } from 'levelEditor/selectors';
import MissingTile from './missing';
import BlankTile from './blank';
import FloorTile from './floor';
import GlassTile from './glass';
import BlockTile from './block';
import RainbowBlockTile from './rainbowBlock';
import BlackHoleTile from './blackHole';
import StickySpotTile from './stickySpot';
import LavaTile from './lava';
import SmartBombTile from './smartBomb';
import ColorChangerTile from './colorChanger';
import GravityChangerTile from './gravityChanger';
import CrateTile from './crate';
import BarrierTile from './barrier';

export const Wrapper = styled.div`
  background: ${props => props.theme.backgroundColor};
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  opacity: ${props => (props.fading ? 0 : 1)};
  transition: all ${props => props.moveSpeed}ms linear;

  ${props =>
    props.stuck &&
    css`
      &:after {
        height: 50%;
        width: 50%;
        background: #00f500;
        border-radius: 100%;
        content: '';
        position: absolute;
        top: 50%;
        left: 50%;
        margin-top: -25%;
        margin-left: -25%;
      }
    `};
`;

export const Tile = ({ entity, moveSpeed }) => {
  let DisplayTile;

  switch (entity.entityId) {
    case undefined:
      DisplayTile = MissingTile;
      break;
    case ENTITIES.NONE:
      DisplayTile = BlankTile;
      break;
    case ENTITIES.FLOOR:
      DisplayTile = FloorTile;
      break;
    case ENTITIES.GLASS:
      DisplayTile = GlassTile;
      break;
    case ENTITIES.BLOCK:
      DisplayTile = BlockTile;
      break;
    case ENTITIES.RAINBOW_BLOCK:
      DisplayTile = RainbowBlockTile;
      break;
    case ENTITIES.BLACK_HOLE:
      DisplayTile = BlackHoleTile;
      break;
    case ENTITIES.STICKY_SPOT:
      DisplayTile = StickySpotTile;
      break;
    case ENTITIES.LAVA:
      DisplayTile = LavaTile;
      break;
    case ENTITIES.SMART_BOMB:
      DisplayTile = SmartBombTile;
      break;
    case ENTITIES.COLOR_CHANGER:
      DisplayTile = ColorChangerTile;
      break;
    case ENTITIES.GRAVITY_CHANGER:
      DisplayTile = GravityChangerTile;
      break;
    case ENTITIES.CRATE:
      DisplayTile = CrateTile;
      break;
    case ENTITIES.BARRIER:
      DisplayTile = BarrierTile;
      break;
    default:
      DisplayTile = MissingTile;
  }

  return (
    <Wrapper
      stuck={entity.stuck}
      shrinking={entity.shrinking}
      fading={entity.fading}
      moveSpeed={moveSpeed}
    >
      <DisplayTile {...entity} moveSpeed={moveSpeed} />
    </Wrapper>
  );
};

Tile.defaultProps = {
  moveSpeed: 0,
};

Tile.propTypes = {
  entity: PropTypes.shape({
    stuck: PropTypes.bool,
    shrinking: PropTypes.bool,
    fading: PropTypes.bool,
  }).isRequired,
  moveSpeed: PropTypes.number,
};

const mapStateToProps = (state, props) => ({
  entity: getEntityForTileId(state, props.tileId),
});

export default connect(mapStateToProps)(Tile);
