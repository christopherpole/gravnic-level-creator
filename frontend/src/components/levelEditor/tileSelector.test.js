import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { spy } from 'sinon';

import tiles from '../../config/tiles.json';
import { TileSelector, TileWrapper } from './tileSelector';

configure({ adapter: new Adapter() });

const props = {
  selectTileAction: () => {},
  selectedTileId: null,
};

describe('<TileSelector />', () => {
  it('Renders without exploding', () => {
    const tileSelector = shallow(<TileSelector {...props} />);

    expect(tileSelector).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const tileSelector = shallow(<TileSelector {...props} />);

    expect(toJson(tileSelector)).toMatchSnapshot();
  });

  it('Matches the current snapshot if a tile is selected', () => {
    const tileSelector = shallow(<TileSelector {...props} selectedTileId={3} />);

    expect(toJson(tileSelector)).toMatchSnapshot();
  });

  it('Fires the select tile action when clicking on a tile', () => {
    const tileClickSpy = spy();
    const tileSelector = shallow(<TileSelector {...props} selectTileAction={tileClickSpy} />);
    const tile = tileSelector.find(TileWrapper).at(3);
    tile.simulate('click');

    expect(tileClickSpy.calledOnce).toBe(true);
    expect(tileClickSpy.calledWith(tiles[3].id)).toBe(true);
  });
});
