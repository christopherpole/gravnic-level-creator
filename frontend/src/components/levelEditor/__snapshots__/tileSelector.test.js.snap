// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileSelector /> Matches the current snapshot 1`] = `
<styled.div
  id="tile-selector"
>
  <styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="0"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={0}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="1"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={1}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="2"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={2}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="3"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={3}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="4"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={4}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="5"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={5}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="6"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={6}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="7"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={7}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="8"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={8}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`<TileSelector /> Matches the current snapshot if a tile is selected 1`] = `
<styled.div
  id="tile-selector"
>
  <styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="0"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={0}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="1"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={1}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="2"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={2}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={true}
      key="3"
      onClick={[Function]}
    >
      <Tile
        isSelected={true}
        tileId={3}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="4"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={4}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="5"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={5}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="6"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={6}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="7"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={7}
      />
    </styled.div>
    <styled.div
      className="tile"
      isSelected={false}
      key="8"
      onClick={[Function]}
    >
      <Tile
        isSelected={false}
        tileId={8}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
